core/src/main/java/com/amplifyframework/core/model/ModelProvider.java

```diff
@@ -33,4 +38,34 @@
      * @return the version string of the models.
      */
     String version();
+
+    /**
+     * A default method TODO.
+     * @return the map of model name to schema of all the models.
+     */
+    default Map<String, ModelSchema> modelSchemas() {
+        final Map<String, ModelSchema> modelSchemaMap = new HashMap<>();
+        for (Class<? extends Model> modelClass : models()) {
+            try {
+                final String modelClassName = modelClass.getSimpleName();
+                final ModelSchema modelSchema = ModelSchema.fromModelClass(modelClass);
+                modelSchemaMap.put(modelClassName, modelSchema);
+            } catch (AmplifyException exception) {
+                exception.printStackTrace();
+            }
+        }
+        return modelSchemaMap;
+    }
+
+    /**
+     * A default method TODO.
+     * @return a set of all model names.
+     */
+    default Set<String> modelNames() {
+        final Set<String> modelNames = new HashSet<>();
+        for (Class<? extends Model> modelClass : models()) {
+            modelNames.add(modelClass.getSimpleName());
+        }
+        return modelNames;
+    }
```

Comment:
Backfill `ModelProvider` to maintain backwards compatibility with android users.